core/src/main/scala/io/snappydata/SnappyDaemons.scala

```diff
@@ -50,25 +50,22 @@ object SnappyAnalyticsService extends Logging {
 				.getOrElse(DEFAULT_ANALYTICS_SERVICE_INTERVAL).toString
 		connProperties =
 				ExternalStoreUtils.validateAndGetAllProps(sc, mutable.Map.empty[String, String])
-		getScheduledExecutor.scheduleWithFixedDelay(
-			getTotalMemoryUsagePerTable, ZERO,
-			delayInMillisconds.toLong, TimeUnit.MILLISECONDS)
-	}
+		if (analyticsExecutor == null || analyticsExecutor.isShutdown) {
+			analyticsExecutor =	newDaemonSingleThreadScheduledExecutor("SnappyAnalyticsService")
+			analyticsExecutor.scheduleWithFixedDelay(
+				getTotalMemoryUsagePerTable, ZERO,
+				delayInMillisconds.toLong, TimeUnit.MILLISECONDS)
+		}
```

Comment:
Actually we should be very averse to adding new threads. There are already tons of them in Gem/GemXD.

Many threads can be reused for this. For example, the stats collection thread or ConnectionSignaller ... We can change the name of latter to be a bit more generic, for example. Alternatively callbacks in stats collection thread should be fine too -- it is reasonable to expect stats to be enabled to get proper values for sizeInBytes (if it is not then it should fallback to default and not return zero).

Second important point is that we already have similar stats collected per node in TableMBeanBridge.memoryAnalytics. I think a function message to just collect and aggregate those stats will be best. Unfortunately the MBean aggregation itself uses a convoluted route of separate replicate region per node for some strange reason (so all member-specific replicated regions have to be created by JMX manager to get data from all). Writing a function message will be so much straightforward. This will avoid an expensive walk-through in every 5 secs as well as query parsing/execution.

Ultimately we should be keeping those stats actively in region itself. BucketRegions already have a bytesInMemory and numOverflowBytesOnDisk though their accuracy needs to be checked (not that we need very accurate results for sizeInBytes, but if there is an error in increment/decrement it can lead to compounding errors). Or we might keep something in GemFireContainer if required, which might be for replicated tables in any case. Likewise for indexes going forward. This will avoid the overhead of having to walk-through the entire table/index which can also help improve SYS.MEMORYANALYTICS itself. We will need to compare the results from the two to ensure there is no error in bookkeeping.

Of course, this can be addressed after this PR. For this PR had discussed with Soubhik and we agree that default for DEFAULT_ANALYTICS_SERVICE_INTERVAL can be increased to 10secs.